src/Codeception/Module/Db.php

```diff
@@ -639,6 +638,22 @@ public function _cleanup($databaseKey = null, $databaseConfig = null)
         }
     }
 
+    /**
+     * @param  array  $databaseConfig
+     * @param  string $databaseKey
+     * @return bool
+     */
+    protected function shouldCleanup($databaseConfig, $databaseKey)
+    {
+        // If using populator and it's not empty, clean up regardless
+        if (!empty($databaseConfig['populator'])) {
+            return true;
+        }
+
+        // If no sql dump for $databaseKey or sql dump is empty, don't clean up
+        return !empty($this->databasesSql[$databaseKey]);
```

Comment:
`return !empty($databaseConfig['populator']) || !empty($this->databasesSql[$databaseKey])`
You explain the logic in comments, so I see no issue with combining this code; it's not a complicated condition.